java/client/src/main/java/glide/internal/GlideCoreClient.java

```diff
@@ -304,6 +304,28 @@ public CompletableFuture<String> updateConnectionPassword(
         return future;
     }
 
+    /** Refresh IAM token (for compatibility with CommandManager) */
+    public CompletableFuture<String> refreshIamToken() {
+        long handle = nativeClientHandle.get();
+        if (handle == 0) {
+            CompletableFuture<String> f = new CompletableFuture<>();
+            f.completeExceptionally(new glide.api.models.exceptions.ClosingException("Client is closed"));
+            return f;
+        }
+
+        CompletableFuture<String> future = new CompletableFuture<>();
+        long correlationId;
+        try {
+            correlationId = AsyncRegistry.register(future, this.maxInflightRequests, handle);
+        } catch (glide.api.models.exceptions.RequestException e) {
+            future.completeExceptionally(e);
+            return future;
+        }
+
+        GlideNativeBridge.refreshIamToken(handle, correlationId);
+        return future;
+    }
```

Comment:
Nit. Could we just have one `CompletableFuture` constructor call..?

```suggestion
    public CompletableFuture<String> refreshIamToken() {
        CompletableFuture<String> future = new CompletableFuture<>();
        
        long handle = nativeClientHandle.get();
        if (handle == 0) {
            future.completeExceptionally(new glide.api.models.exceptions.ClosingException("Client is closed"));
            return future;
        }

        long correlationId;
        try {
            correlationId = AsyncRegistry.register(future, this.maxInflightRequests, handle);
        } catch (glide.api.models.exceptions.RequestException e) {
            future.completeExceptionally(e);
            return future;
        }

        GlideNativeBridge.refreshIamToken(handle, correlationId);
        return future;
    }
```